src/System.Linq.Parallel/tests/ExchangeTests.cs

```diff
@@ -10,183 +11,177 @@ namespace Test
+        [MemberData("MergeData", (object)(new int[] { 1024 * 4, 1024 * 1024 }))]
+        [MemberData("FailingMergeData", (object)(new int[] { 1024 * 4, 1024 * 1024 }))]
+        public static void Merge_Ordered_Pipelining_Longrunning(Labeled<ParallelQuery<int>> labeled, int count, ParallelMergeOptions options)
         {
-            PartitioningTestCore(true, 2, 0, 900);
-            PartitioningTestCore(false, 2, 0, 900);
+            Merge_Ordered_Pipelining(labeled, count, options);
         }
 
-        private static void PartitioningTestCore(bool stripedPartitioning, int partitions, int minLen, int maxLen)
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 2 }), MemberType = typeof(UnorderedSources))]
+        public static void Merge_ArgumentException(Labeled<ParallelQuery<int>> labeled, int count)
         {
-            for (int len = minLen; len < maxLen; len++)
-            {
-                int[] arr = Enumerable.Range(0, len).ToArray();
-                IEnumerable<int> query;
+            ParallelQuery<int> query = labeled.Item;
 
-                if (stripedPartitioning)
-                {
-                    query = arr.AsParallel().AsOrdered().WithDegreeOfParallelism(partitions).Take(len).Select(i => i);
-                }
-                else
-                {
-                    query = arr.AsParallel().AsOrdered().WithDegreeOfParallelism(partitions).Select(i => i);
-                }
-
-                if (!arr.SequenceEqual(query))
-                {
-                    Console.WriteLine("PartitioningTest: {0}, {1}, {2}, {3}", stripedPartitioning, partitions, minLen, maxLen);
-                    Assert.True(false, string.Format("  ** FAILED: incorrect output for array of length {0}", len));
-                }
-            }
+            Assert.Throws<ArgumentException>(() => query.WithMergeOptions((ParallelMergeOptions)4));
         }
 
         [Fact]
-        public static void OrderedPipeliningTest()
+        public static void Merge_ArgumentNullException()
         {
-            OrderedPipeliningTest1(4, true);
-            OrderedPipeliningTest1(4, false);
-            OrderedPipeliningTest1(10007, true);
-            OrderedPipeliningTest1(10007, false);
-
-            OrderedPipeliningTest2(true);
-            OrderedPipeliningTest2(false);
+            Assert.Throws<ArgumentNullException>(() => ((ParallelQuery<int>)null).WithMergeOptions(ParallelMergeOptions.AutoBuffered));
         }
 
         /// <summary>
-        /// Checks whether an ordered pipelining merge produces the correct output.
+        /// The plinq chunk partitioner calls takes an IEnumerator over the source, and disposes the enumerator when it is
+        /// finished.
+        /// If an exception occurs, the calling enumerator disposes the enumerator... but then other callers generate ODEs.
+        /// These ODEs either should not occur (preferred), or should not get into the aggregate exception.
```

Comment:
I don't understand this exception.  Are you commenting about what you believe to be a bug in the implementation?  If so, why is this in the XML comment for a test?